### ISSUE TITLE
[BUG] Node driver and cloud provider enhancement of RKE2 cluster with external Harvester cluster

### ISSUE THREAD

**Issue by janeczku** [harvester/harvester#1244]
**To Reproduce**
1. Create a Harvester cloud credential choosing "External Harvester" and providing valid Kubeconfig
2. Navigate to the form to create Harvester RKE2 cluster
3. Pick the credential created above
4. Fields are **not** populated with available options
5. Harvester cloud provider is not working as expected as described on https://github.com/rancher/rancher/issues/34969

**Expected behavior**
- Fields are populated with available options from Harvester (namespaces, networks, etc)
- Cloud provider integration should also works with external Harvester cluster

** Actual behavior**
Fields are **not** populated with available options

**Comment by futuretea** [comment]
Only "Imported Harvester" support "Fields are populated with available options from Harvester (namespaces, networks, etc)"

**Comment by yasker** [comment]
@futuretea Is there any specific reason why field population can only be supported with imported cluster?

**Comment by guangbochen** [comment]
@yasker IIRC we have discussed this previously, the front-end will need to implement the `kubeclient` to call the external Harvester server and there could be a potential certificate and CROS issue upon the environment setup, therefore we have decided just to ask the user to fill those filed manually for now if they choose to use external Harvester. But can be improved in the future.

Imported Harvester cluster is always the first option to use.
